pkg/mount/mount_test.go

```diff
@@ -79,6 +85,22 @@ var mountinfo = `39 21 0:36 / /tmp rw shared:91 -
 255 193 179:1 / /prefix/stage1/rootfs/opt/stage2/busybox/rootfs/rootfs/mnt/unknown ro,relatime -
 `
 
+// safeOrder checks for transitivity and (un)mount order sanity
```

Comment:
Hmm .. I am aware that this method has simply been copied, but I am unsure about the whole implementation. 

Checking for transitivity here is O(n^3), and fails in my playground test for the last element https://play.golang.org/p/15NbcNUUPH.

Doesn't this boil down to check if the given `Mounts` must be sorted in ascending order according to the mount depth?

If so, why not simply copy the list `m`, sort it, and compare it (see the above playround `safeOrder2`)?